### PR TITLE
Add HEAD support to download and publish (#14)

### LINKED ISSUE #14: Support HEAD request for the download specific crate route
Sending HEAD request to `https://crates.io/api/v1/crates/serde/1.0.0/download` is supported

it would be great supporting this too

### PR DESCRIPTION
**Pull request by rluvaton**
- [x] wait for #18 to merge before this

Fix #14

### CODE REVIEW

File: src/serve.rs [hunk 1]
```diff
@@ -154,15 +156,32 @@ pub async fn serve(root: &Path, binding: impl Into<ServerBinding>, server_addr:
...
         .with(warp::trace::request());
-    let download = warp::get()
-        .and(warp::path("api"))
+    let download = warp::path("api")
         .and(warp::path("v1"))
         .and(warp::path("crates"))
         .and(warp::path::param())
         .and(warp::path::param())
         .and(warp::path("download"))
+        .and(
```
Comment (TalRoni) [inline src/serve.rs:182]: Hi @rluvaton,
Thank you for your contribution to this repository.
If I'm not mistaken, in crates.io, the HEAD request is used as a fallback for all API calls. (see https://github.com/rust-lang/crates.io/blob/main/src/router.rs#L172) We might prefer to implement this behavior. What do you think?

Comment (rluvaton) [reply src/serve.rs:182]: Is it possible in warp?

Comment (rluvaton) [reply src/serve.rs:182]: Added manually for publish endpoint as well

Comment (rluvaton) [reply src/serve.rs:182]: ping

**Review by TalRoni — APPROVED** [review]

**Pull request merged**